src/interpolation_transfer.jl

```diff
@@ -85,56 +86,83 @@ function interpolate_kpoint(data_in::AbstractVecOrMat, kpoint_in::Kpoint, kpoint
-
 """
-Compute indices of the vectors in G_vectors(kpt_in) that are also present
-in G_vectors(kpt_out).
+Compute the index mapping between two bases. Returns two arrays
+`idcs_in` and `idcs_out` such that `ψkout[idcs_out] = ψkin[idcs_in]` does
+the transfer from `ψkin` (defined on `basis_in` and `kpt_in`) to `ψkout`
+(defined on `basis_out` and `kpt_out`).
 """
-function transfer_blochwave_mapping(basis_in::PlaneWaveBasis{T}, kpt_in::Kpoint,
-                                    basis_out::PlaneWaveBasis{T}, kpt_out::Kpoint) where T
+function transfer_mapping(basis_in::PlaneWaveBasis{T},  kpt_in::Kpoint,
+                          basis_out::PlaneWaveBasis{T}, kpt_out::Kpoint) where T
+    idcs_in  = 1:length(G_vectors(kpt_in))  # All entries from idcs_in
+    kpt_in == kpt_out && return idcs_in, idcs_in
+
     # Get indices of the G vectors of the old basis inside the new basis.
-    idcsk_out = index_G_vectors.(Ref(basis_out), G_vectors(kpt_in))
-    # In the case where G_vectors(basis_in.kpoints[ik]) are biggers than vectors
+    idcs_out = index_G_vectors.(Ref(basis_out), G_vectors(kpt_in))
+
+    # In the case where G_vectors(basis_in.kpoints[ik]) are bigger than vectors
     # in the fft_size box of basis_out, we need to filter out the "nothings" to
-    # make sure that the indices linearization works. It is not an issue to
+    # make sure that the index linearization works. It is not an issue to
     # filter these vectors as this can only happen if Ecut_in > Ecut_out.
-    filter!(!isnothing, idcsk_out)
-    idcsk_out = getindex.(Ref(LinearIndices(basis_out.fft_size)), idcsk_out)
+    if any(isnothing, idcs_out)
+        idcs_in  = idcs_in[idcs_out .!= nothing]
+        idcs_out = idcs_out[idcs_out .!= nothing]
+    end
+    idcs_out = getindex.(Ref(LinearIndices(basis_out.fft_size)), idcs_out)
 
     # Map to the indices of the corresponding G-vectors in G_vectors(kpt_out)
     # this array might contains some nothings if basis_out has less G_vectors
     # than basis_in at this kpoint
-    indexin(idcsk_out, kpt_out.mapping)
+    idcs_out = indexin(idcs_out, kpt_out.mapping)
+    if any(isnothing, idcs_out)
+        idcs_in  = idcs_in[idcs_out .!= nothing]
+        idcs_out = idcs_out[idcs_out .!= nothing]
+    end
+
+    idcs_in, idcs_out
 end
 
+
+"""
+Return a sparse matrix that maps quantities given in the
+`basis_in` basis and on `kpt_in` to quantities given in the `basis_out` basis
+and on `kpt_out`.
+"""
+function transfer_matrix(basis_in::PlaneWaveBasis{T}, kpt_in::Kpoint,
+                         basis_out::PlaneWaveBasis{T}, kpt_out::Kpoint) where T
+    idcs_in, idcs_out = transfer_mapping(basis_in, kpt_in, basis_out, kpt_out)
+    sparse(idcs_out, idcs_in, true)
+end
+
+
+"""
+Return a list of sparse matrices (one per `k`-Point) that map quantities given in the
+`basis_in` basis to quantities given in the `basis_out` basis.
+"""
+function transfer_matrix(basis_in::PlaneWaveBasis{T}, basis_out::PlaneWaveBasis{T}) where T
```

Comment:
I would remove that method. It doesn't mesh with its name (since it's matrices) and it's trivial

Reply:
Well it is a matrix (blocks of a a matrix) and most people will likely need that function and not the individual blocks (to transfer between bases). We can argue about names, but I would say this function and the basis/basis version of `transfer_blochwave` sit pretty much at the same level of "trivial". If we want one, we likely want the other.